doc/rst/language/spec/procedures.rst

```diff
@@ -576,70 +576,93 @@ Abstract Intents
-=================== ================ ======================= ====================================================
-
 .. _The_Const_Intent:
 
 The Const Intent
 ^^^^^^^^^^^^^^^^
 
-The ``const`` intent specifies the intention that the function will not
-and cannot modify the formal argument within its dynamic scope. Whether
-the actual argument will be passed by ``const in`` or ``const ref``
-intent depends on its type. In general, small values, such as scalar
-types, will be passed by ``const in``; while larger values, such as
-domains and arrays, will be passed by ``const ref`` intent. The
-:ref:`Abstract_Intents_Table` earlier in this sub-section lists the
-meaning of the const intent for each type.
+The ``const`` intent specifies that the function will not and cannot
+modify the formal argument within its dynamic scope. Whether ``const``
+is interpreted as ``const in`` or ``const ref`` intent depends on the
+argument type.  Generally, small values, such as scalar types, will be
+passed by ``const in``; while larger values, such as domains and
+arrays, will be passed by ``const ref`` intent. The
+:ref:`Abstract_Intents_Table` below lists the meaning of the ``const``
+intent for each type.
 
 .. _The_Default_Intent:
 
 The Default Intent
 ^^^^^^^^^^^^^^^^^^
 
-When no intent is specified for a formal argument, the *default intent*
-is applied. It is designed to take the most natural/least surprising
-action for the argument, based on its type.
-The :ref:`Abstract_Intents_Table` earlier in this sub-section lists the
-meaning of the default intent for each type.
+When no intent is specified for a formal argument, the *default
+intent* is applied.  It is designed to take the most natural/least
+surprising action for the argument, based on its type.  In practice,
+this is ``const`` for most types (as defined by
+:ref:`The_Const_Intent`) to avoid surprises for programmers coming
+from languages where everything is passed by ``in`` or ``ref`` intent
+by default.  Exceptions are made for types where modification is
+considered part of its nature, such as types used for synchronization
+(like ``atomic``) and arrays.
 
 Default argument passing for tuples generally matches the default
 argument passing strategy that would be applied if each tuple element
 was passed as a separate argument. See :ref:`Tuple_Argument_Intents`.
 
+The :ref:`Abstract_Intents_Table` that follows defines the default
+intent for each type.
+
+.. _Abstract_Intents_Table:
+
+Abstract Intents Table
+^^^^^^^^^^^^^^^^^^^^^^
+
+The following table summarizes what these abstract intents mean for each
+type:
+
+.. table::
+    :widths: 28 18 22 32
+
+    +-------------------------+----------------------+-------------------------+------------------------------------------------------+
+    |                         | ``const`` intent     |                         |                                                      |
```

Comment:
Is there a way to get this to 80 columns and still look the same in the HTML? As it stands I think it'll be hard to read in the .rst (but that's not a huge problem if it's a real challenge to change). I'm not familiar with these table width features but I would imagine that the `:widths:` feature might control the HTML size while ignoring the length of the lines etc in the .rst?

Reply:
There probably is, but I don't know that it's worth it (and I say that as a firm believer in the 80-column format).  Specifically, I'm not aware of a way to break the cross-references or code-formatted entries across multiple lines (or, even if there were one, I'm not sure that it would be a good idea from an rst-readability perspective), making the shortest possible line ~105 characters:

```
    | synchronization |	``const ref`` |	``const ref`` |	:ref:`Default_Intent_for_Arrays_and_Record_this` |
```

The other approach would be that I don't think the column separators actually need to line up in the ascii, but mis-aligning them would again make the rst hard to read.  So I threw 80 columns under the bus and tried instead to arrange the rst file so that it used linefeeds in reasonable ways while trying not to orphan stuff unnecessarily.